### PR TITLE
Support for a `next` param to redirect after a successful login

### PR DESCRIPTION
**Pull request by marcospri**
The goal here is to be able to go, unauthenticated to http://localhost:8001/admin/instance/id/101/, get an auth prompt and be redirected there instead of to the root.

We use the "state" param to hold the target URL across the whole transaction. As a result state becomes a base64 json that holds the existing nonce parameter too.

Found some out of date things in this repo and fixed them along the way in separate commits.

# Testing

- Checkout LMS's branch `google-auth-next` (https://github.com/hypothesis/lms/pull/4185)

- Checkout this branch and point LMS to it:

```
.tox/dev/bin/pip install -e /home/marcos/hypo/pyramid-googleauth
```

- Go to localhost:8001/admin/instance/id/101/

Complete the oauth flow and end  up there (instead of in `/admin`)

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -11,8 +11,7 @@ load-plugins=pylint.extensions.bad_builtin,
              pylint.extensions.redefined_variable_type,
 
 [MESSAGES CONTROL]
-disable=bad-continuation,
-        missing-docstring,
+disable=missing-docstring,
```
Comment (marcospri) [inline .pylintrc:14]: pylint upgrade

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+[tool.hdev]
```
Comment (marcospri) [inline pyproject.toml:1]: Without this all hdev commands (that we use in make for example) were broken.

File: tests/unit/pyramid_googleauth/services/google_auth_test.py [hunk 3]
```diff
@@ -28,6 +28,7 @@ def test_it_allows_non_https_redirect_uris_for_dev(
 
         assert bool(os.environ.get("OAUTHLIB_INSECURE_TRANSPORT")) == insecure
 
+    @pytest.mark.usefixtures("_get_nonce", "_decode_state", "base64")
```
Comment (marcospri) [inline tests/unit/pyramid_googleauth/services/google_auth_test.py:31]: The test in this file are a bit difficult to refactor as they assert calls on the own service's private methods.

File: tox.ini [hunk 4]
```diff
@@ -23,7 +23,7 @@ passenv =
     dist: BUILD
 deps =
     dev: -r requirements-dev.txt
-    {tests,lint,functests}: .[tests]
+    {tests,lint,functests}: -e .[tests]
```
Comment (marcospri) [inline tox.ini:26]: Without this the test were using the installed version of the package, not the current workdir.

File: src/pyramid_googleauth/security.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 class GoogleSecurityPolicy:
     @property
-    @lru_cache(maxsize=None)
+    @lru_cache(maxsize=1)
```
Comment (jon-betts) [inline src/pyramid_googleauth/security.py:8]: Ooof... yeah. Good catch.

**Review by jon-betts — APPROVED** [review]
This looks like a pretty sensible way of doing it. Is this the kind of "standard" for this? It seems like  a thing lots of people would want to do.

**Pull request merged**